File: boudicca.base/semantic-conventions/src/main/java/base/boudicca/EventCategory.java
package base.boudicca;

import java.util.Set;

public enum EventCategory {

    MUSIC(Set.of("konzert", "concert", "alternative", "singer/songwriter", "party", "songwriter/alternative", "musik")),
    ART(Set.of("kabarett", "theater", "wissenskabarett", "provinzkrimi", "comedy", "figurentheater", "film", "visual comedy", "tanz", "performance", "musiklesung", "literatur", "museen", "museum", "ausstellung", "musical")),
    TECH(Set.of("techmeetup"));

    public final Set<String> types;

    EventCategory(Set<String> types) {
        this.types = types;
    }

    public static EventCategory getForType(String type) {
        if (type == null) {
            return null;
        }

        String lowerType = type.toLowerCase();
        for (EventCategory category : EventCategory.values()) {
            for (String categoryType : category.types) {
                if (lowerType.contains(categoryType)) {
                    return category;
                }
            }
        }
        return null;
    }
}

File: boudicca.base/common-model/src/main/kotlin/base/boudicca/model/EventCategory.kt
package base.boudicca.model

import java.util.Locale

/**
 * current event categories we support, with all their known event types
 */
enum class EventCategory(val types: Set<String>) {
    // OTHER is a special cases needing special care
    OTHER(emptySet()),
    MUSIC(setOf("konzert", "concert", "alternative", "singer/songwriter", "party", "songwriter/alternative", "musik")),
    ART(
        setOf(
            "kabarett",
            "theater",
            "wissenskabarett",
            "provinzkrimi",
            "comedy",
            "figurentheater",
            "film",
            "visual comedy",
            "tanz",
            "performance",
            "musiklesung",
            "literatur",
            "museen",
            "museum",
            "ausstellung",
            "musical",
            "oper",
            "brauchtum",
            "schauspiel"
        )
    ),
    TECH(setOf("techmeetup", "technology", "technologie", "chaosevent")),
    SPORT(setOf("sport", "football", "soccer", "actionsport", "esports"));

    companion object {
        fun getForType(type: String?): EventCategory? {
            if (type == null) {
                return null
            }
            val lowerType = type.lowercase(Locale.getDefault())
            for (category in entries) {
                for (categoryType in category.types) {
                    if (lowerType.contains(categoryType)) {
                        return category
                    }
                }
            }
            return null
        }
    }
}
